File: rust/src/models/batch_job_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{batch_job_response::BatchJobResponse, pagination_response::PaginationResponse};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct BatchJobListResponse {
    pub data: Vec<BatchJobResponse>,

    pub pagination_meta: PaginationResponse,
}

impl BatchJobListResponse {
    pub fn new(data: Vec<BatchJobResponse>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/batchJobListResponse.ts
// this file is @generated
import { type BatchJobResponse, BatchJobResponseSerializer } from "./batchJobResponse";
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";

export interface BatchJobListResponse {
  data: BatchJobResponse[];

  paginationMeta: PaginationResponse;
}

export const BatchJobListResponseSerializer = {
  _fromJsonObject(object: any): BatchJobListResponse {
    return {
      data: object["data"].map((item: any) =>
        BatchJobResponseSerializer._fromJsonObject(item)
      ),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: BatchJobListResponse): any {
    return {
      data: self.data.map((item: any) => BatchJobResponseSerializer._toJsonObject(item)),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
